src/Validation/Validator/StringValidator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ScrumWorks\OpenApiSchema\Validation\Validator;
+
+use RuntimeException;
+use ScrumWorks\OpenApiSchema\Validation\BreadCrumbPath;
+use ScrumWorks\OpenApiSchema\Validation\Result\ValidationResultBuilderFactoryInterface;
+use ScrumWorks\OpenApiSchema\Validation\Result\ValidationResultBuilderInterface;
+use ScrumWorks\OpenApiSchema\ValueSchema\StringSchema;
+
+final class StringValidator extends AbstractValidator
+{
+    private StringSchema $schema;
+
+    public function __construct(
+        ValidationResultBuilderFactoryInterface $validationResultBuilderFactory,
+        StringSchema $schema
+    ) {
+        parent::__construct($validationResultBuilderFactory, $schema);
+
+        $this->schema = $schema;
+    }
+
+    protected function doValidation(
+        ValidationResultBuilderInterface $resultBuilder,
+        $data,
+        BreadCrumbPath $breadCrumbPath
+    ): void {
+        if (! $this->validateNullable($resultBuilder, $data, $breadCrumbPath)) {
+            return;
+        }
+
+        if (! \is_string($data)) {
+            $resultBuilder->addTypeViolation('string', $breadCrumbPath);
+            return;
+        }
+
+        $strlen = $this->schema->getFormat() === 'binary' ? \strlen($data) : \mb_strlen($data);
+
+        if (($minLength = $this->schema->getMinLength()) !== null && $minLength >= $strlen) {
+            $resultBuilder->addMinLengthViolation($minLength, $breadCrumbPath);
+        }
+        if (($maxLength = $this->schema->getMaxLength()) !== null && $maxLength <= $strlen) {
+            $resultBuilder->addMaxLengthViolation($maxLength, $breadCrumbPath);
+        }
+        if (($pattern = $this->schema->getPattern()) !== null && ! \preg_match("~${pattern}~", $data)) {
+            $resultBuilder->addPatternViolation($breadCrumbPath);
+        }
+        if (($format = $this->schema->getFormat()) !== null && ! \preg_match($this->formatToPattern($format), $data)) {
+            $resultBuilder->addFormatViolation($breadCrumbPath);
+        }
+    }
+
+    private function formatToPattern(string $format): string
+    {
+        switch ($format) {
+            case 'binary': return '~.*~';
+            case 'date': return '~[0-9]{4}-[0-9]{2}-[0-9]{2}~';
+            case 'date-time': return '~[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]+(Z|([+-][0-9]{2}:[0-9]{2})))?~';
+            default: throw new RuntimeException("Not supported format '${format}'.");
```

Comment:
OpenAPI primo definuje tyto formaty:
date – full-date notation as defined by RFC 3339, section 5.6, for example, 2017-07-21
date-time – the date-time notation as defined by RFC 3339, section 5.6, for example, 2017-07-21T17:32:28Z
password – a hint to UIs to mask the input
byte – base64-encoded characters, for example, U3dhZ2dlciByb2Nrcw==
binary – binary data, used to describe files (see Files below)

ale dovoluje definovat nove, takze (jako treba email, ipv6, ...) - takze je otazka - jak tam pripane dostanu svuj vlastni format?

Reply:
Neresil jsem, az bude potreba, tak se tahle zodpovednost vytahne do vlastniho objektu

Reply:
ok, jen ze by na to bylo dobre myslet, ale tak minimalne ty definovane z OpenAPI bych tam pridal s tim ze pro ne hod specialni hlasku - jakoze ok, existuji, ale zatim nejsou implementovane... - at se na ne nezapomene jestli nejsi proti